python/GafferUI/SpreadsheetUI.py

```diff
@@ -0,0 +1,1560 @@
+		self.__emitModelReset()
+		self.headerDataChanged.emit( QtCore.Qt.Horizontal, 0, self.columnCount() )
+
+	def __columnRemoved( self, cellsPlug, cellPlug ) :
+
+		## \todo Is there any benefit in finer-grained signalling?
+		self.__emitModelReset()
+
+	def __plugDirtied( self, plug ) :
+
+		if not self.__rowsPlug.isAncestorOf( plug ) :
+			return
+
+		parentPlug = plug.parent()
+		if isinstance( parentPlug, Gaffer.Spreadsheet.RowPlug ) and plug.getName() == "enabled" :
+			# Special case. The enabled plug affects the flags for the entire row.
+			# Qt doesn't have a mechanism for signalling flag changes, so we emit `dataChanged`
+			# instead, giving our delegate the kick it needs to redraw.
+			rowIndex = parentPlug.parent().children().index( parentPlug )
+			self.dataChanged.emit( self.index( rowIndex, 0 ), self.index( rowIndex, self.columnCount() - 1 ) )
+			return
+
+		# Regular case. Emit `dataChanged` for just this one plug.
+		index = self.indexForPlug( plug )
+		if index.isValid() :
+			self.dataChanged.emit( index, index )
+
+	def __plugMetadataChanged( self, nodeTypeId, plugPath, key, plug ) :
+
+		if plug is None :
+			return
+
+		index = self.indexForPlug( plug )
+		if not index.isValid() :
+			return
+
+		if key == "spreadsheet:columnLabel" :
+			self.headerDataChanged.emit( QtCore.Qt.Horizontal, index.column(), index.column() )
+		elif Gaffer.MetadataAlgo.readOnlyAffectedByChange( key ) :
+			# Read-only metadata is actually reflected in `flags()`, but there's no signal to emit
+			# when they change. Emitting `dataChanged` is enough to kick a redraw off.
+			self.dataChanged.emit( index, index )
+
+	def __contextChanged( self, context, key ) :
+
+		if key.startswith( "ui:" ) :
+			return
+
+		self.__emitModelReset()
+
+	def __emitModelReset( self ) :
+
+		self.beginResetModel()
+		self.endResetModel()
+
+	def __displayRoleValue( self, index ) :
+
+		plug = self.valuePlugForIndex( index )
+		if isinstance( plug, Gaffer.BoolPlug ) :
+			# Dealt with via CheckStateRole
+			return None
+
+		try :
+			with self.__context :
+				currentPreset = Gaffer.NodeAlgo.currentPreset( plug )
+				if currentPreset is not None :
+					return currentPreset
+				if isinstance( plug, Gaffer.NameValuePlug ) :
+					return plug["enabled"].getValue(), plug["value"].getValue()
+				else :
+					return plug.getValue()
+		except :
+			return None
+
+	def __formatValue( self, value, forToolTip = False ) :
+
+		if isinstance( value, str ) :
+			return value
+		elif isinstance( value, ( int, float ) ) :
+			return GafferUI.NumericWidget.valueToString( value )
+		elif isinstance( value, ( imath.V2i, imath.V2f, imath.V3i, imath.V3f ) ) :
+			return ", ".join( GafferUI.NumericWidget.valueToString( v ) for v in value )
+		elif isinstance( value, bool ) :
+			return value
+		elif isinstance( value, tuple ) :
+			# NameValuePlug ( enabled, value )
+			s = self.__formatValue( value[1], forToolTip )
+			return "{}{}{}".format(
+				"On" if value[0] else "Off",
```

Comment:
How do you feel about `:` as the delimiter between enabled/value? Just to separate it out from components of the value?

Reply:
I like it : 71b3a21.